common/GL/ContextEGL.cpp

```diff
@@ -318,41 +313,34 @@ namespace GL
 		}
 		configs.resize(static_cast<u32>(num_configs));
 
-		std::optional<EGLConfig> config;
-		for (EGLConfig check_config : configs)
-		{
-			if (CheckConfigSurfaceFormat(check_config))
-			{
-				config = check_config;
-				break;
-			}
-		}
+		auto config = [this, &configs]() {
+			const auto found_config = std::find_if(std::begin(configs), std::end(configs), [&](auto&& check_config) { return CheckConfigSurfaceFormat(check_config); });
```

Comment:
Is there anything preventing you from using `CheckConfigSurfaceFormat` directly instead of wrapping it in a lambda that calls it?

Also you can remove the usage of the `std::optional` completely if you pull the lower `!config.has_value()` check into here

Reply:
I mean like this
```cpp
std::find_if(std::begin(configs), std::end(configs), CheckConfigSurfaceFormat)
```